src/SourceCode.Clay.Collections/CaseInsensitiveSwitchBuilder.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+
+namespace SourceCode.Clay.Collections.Generic
+{
+    /// <summary>
+    /// A dynamic switch for dictionaries with <see cref="System.String"/> keys.
+    /// Uses an case-insensitive (invariant) comparison of key values.
+    /// </summary>
+    /// <typeparam name="TValue">The type of values.</typeparam>
+    internal sealed class CaseInsensitiveSwitchBuilder<TValue> : BaseSwitchBuilder<string, TValue>
+    {
+        public CaseInsensitiveSwitchBuilder(IReadOnlyDictionary<string, TValue> cases)
+            : base(cases)
+        { }
+
+        protected override string NormalizeKey(string key) => key.ToLowerInvariant();
```

Comment:
`ToUpperInvariant()`. Read a few years back that upper-case is optimized in the CLR for some reason.

At some point we will need to figure this out. E.g. Bake in hashcodes in the switch and then use `if` (which is what the JITter does). This will also free us from using switch types that C# supports natively.